woocommerce/wc-admin
# DatePicker: Add url support and launcher

This PR adds:
* A dropdown courtesy of Gutenberg's `<Dropdown />` component
* Url as source of truth, i.e. date parameters `period`, `compare`, and `start` and `end` if the period is custom
* "Update" button is now a link with a constructed url to reflect newly selected parameters

**Note: The date ranges in parenthesis are hard coded strings for now so they won't update**

Designs can be seen in p6riRB-3ch-p2

## Result
### Styled button

![screen shot 2018-05-29 at 3 05 25 pm](https://user-images.githubusercontent.com/1922453/40636147-c36b2102-6351-11e8-8d53-2f73322ce71c.png)

### Expanded view

![screen shot 2018-05-29 at 3 00 04 pm](https://user-images.githubusercontent.com/1922453/40636106-8d3474ee-6351-11e8-9dd5-366c87cd28bb.png)

### Expanded mobile view

![screen shot 2018-05-29 at 3 00 24 pm](https://user-images.githubusercontent.com/1922453/40636107-8fab006c-6351-11e8-8b94-1d71ae605cdf.png)

## Test
* Head to `/wp-admin/admin.php?page=woodash#/analytics`
* See the button reflect default parameters `today` and `previous_period`
* Click the button, see the dropdown expand
* Select new parameters
* Click update
* See the dropdown close
* See the updated parameters reflected in the url
* See the updated parameters reflected in the button labels (except the date ranges)
* If you update with a custom date, parameters `start` and `end` will be added as url parameters
* Ensure behaviour is as expected
* Reduce screen size to see mobile styles

## Next Steps
* `date-utils` to fully parse periods and custom values (include unit tests)
* Tackle the custom date picker 😰

Fixes https://github.com/woocommerce/woo-dash/issues/62
